### PR TITLE
Add validation on action names

### PR DESCRIPTION
**Pull request by cfournie**
Oozie [workflow action names are identifiers](https://github.com/apache/oozie/blob/35db5b31fa0c69680de4fe12eabc6fde818b2c54/client/src/main/resources/oozie-workflow-0.5.xsd#L159), and identifiers must conform to [`([a-zA-Z_]([\-_a-zA-Z0-9])*){1,39}`](https://github.com/apache/oozie/blob/35db5b31fa0c69680de4fe12eabc6fde818b2c54/client/src/main/resources/oozie-workflow-0.5.xsd#L26). This PR adds this enforcement.

review/ @honkfestival @kmtaylor-github

### CODE REVIEW

File: tests/pyoozie/test_builder.py [hunk 1]
```diff
@@ -134,6 +134,29 @@ def test_workflow_builder():
...
+        ).add_action(
+            name='Name with invalid characters',
+            action=Shell(exec_command='echo "test"'),
+            action_on_error=Email(to='person@example.com', subject='Error', body='A bad thing happened'),
+            kill_on_error='Failure message',
+        )
+    assert str(assertion_info.value) == \
+        "Identifier must match ^[a-zA-Z_][\\-_a-zA-Z0-9]{0,38}$, 'Name with invalid characters' does not"
+
+    # Does it raise an exception when you try to add multiple actions?
```
Comment (kmtaylor-github) [inline tests/pyoozie/test_builder.py]: Bonus test? 🙂

Comment (cfournie) [inline tests/pyoozie/test_builder.py]: Ah, yeaaah. That should be a different PR.

File: tests/pyoozie/test_builder.py [hunk 2]
```diff
@@ -134,6 +134,29 @@ def test_workflow_builder():
...
+        WorkflowBuilder(
+            name='descriptive-name'
+        ).add_action(
+            name='Name with invalid characters',
+            action=Shell(exec_command='echo "test"'),
+            action_on_error=Email(to='person@example.com', subject='Error', body='A bad thing happened'),
+            kill_on_error='Failure message',
+        )
+    assert str(assertion_info.value) == \
+        "Identifier must match ^[a-zA-Z_][\\-_a-zA-Z0-9]{0,38}$, 'Name with invalid characters' does not"
```
Comment (honkfestival) [inline tests/pyoozie/test_builder.py]: Isn't it [`{1,39}`](https://github.com/apache/oozie/blob/35db5b31fa0c69680de4fe12eabc6fde818b2c54/client/src/main/resources/oozie-workflow-0.5.xsd#L26)?

Comment (cfournie) [inline tests/pyoozie/test_builder.py]: Yep, note the round brackets in the schema regex (`([a-zA-Z_]([\-_a-zA-Z0-9])*){1,39}`) and the square brackets in the Python implementation.

Comment (kmtaylor-github) [inline tests/pyoozie/test_builder.py]: facepalm-icon why why why would you specify your regex like that?

Comment (cfournie) [inline tests/pyoozie/test_builder.py]: Do you want me to make it closer to the Oozie XML schema's regex?

Comment (kmtaylor-github) [inline tests/pyoozie/test_builder.py]: No, yours is perfectly sensible. Theirs is not.

**Comment by honkfestival** [comment]
LGTM

File: tests/pyoozie/test_builder.py (hunk 2 above)
Comment (cfournie) [inline tests/pyoozie/test_builder.py]: Upon closer inspection, Oozie may have intended to limit identifiers to 39 characters, but because to the `*` they do not. My regex follows the spirit of their regex, but not what it actually restricts.

Should we use their regex with no length limit? Or my more limited regex?

Comment (kmtaylor-github) [inline tests/pyoozie/test_builder.py]: Sigh... I thought it was bad, but apparently it's worse.

I'd vote for respecting the intended (?) 39 char limit to ensure we don't get surprised if they fix the schema in the future.

Comment (honkfestival) [inline tests/pyoozie/test_builder.py]: Yeah, the prudent choice here is the right one.
